### ISSUE TITLE
Blocking get aio_pika

### ISSUE THREAD

**Issue by josephbiko** [mosquito/aio-pika#131]
I expected

await queue.get() 
to be blocking in aio_pika, but even when I don't set the timeout parameter I instantly get an error:

aio_pika.exceptions.QueueEmpty

Anyway to get a blocking get in aio_pika?

**Comment by josephbiko** [comment]
Is there also a way to just wait until a message is available? 
Or do I need to sleep&check until something is available?

**Comment by mosquito** [comment]
You should call consume for the queue. [This example](http://aio-pika.readthedocs.io/en/latest/quick-start.html#simple-consumer) do it easer.

**Comment by mosquito** [comment]
[This example](https://aio-pika.readthedocs.io/en/latest/quick-start.html#asynchronous-message-processing) demonstrated the asyncronous m

**Comment by josephbiko** [comment]
Yes, but this is a normal consumer, I just want to retrieve a single message from the queue

**Comment by mosquito** [comment]
The `basic.get` method marked as synchronous on the [documentation (page 54)](https://www.rabbitmq.com/resources/specs/amqp-xml-doc0-9-1.pdf)

```
Synchronous: Yes; expected response is from method(s) basic.get­ok, basic.get­empty
```

You should use this method on very specific cases.

**Comment by josephbiko** [comment]
Allow me to illustrate,
What i do know is:
` while True:
            msg = await q.get(fail=False)
            if msg:
                break
            await asyncio.sleep(1)`

**Comment by nurettin** [comment]
If we use consume, it will just get messages regardless of whether the consume callback has finished it's work or not. This is undesirable when your consumer can not work concurrently. Ideally, this would do nicely:

    consume(callback, wait_until_callback_finishes_before_receiving_any_more_messages=True)

**Comment by mosquito** [comment]
@nurettin just `set_qos(prefetch_count=1)` and stop consuming before ack

**Comment by nurettin** [comment]
is this your suggestion? @mosquito 

     set_qos(prefetch_count=1)
     consume(cb)

     and in cb: 
         cancel
         process
         ack
         consume(cb)
